src/common/components/Settings.tsx

```diff
@@ -982,18 +1033,87 @@ export function InnerSettings({ onSave }: IInnerSettingsProps) {
                     const { isEnabled: autostartIsEnabled } = await import('tauri-plugin-autostart-api')
                     settings.runAtStartup = await autostartIsEnabled()
                 }
-                setValues(settings)
-                setPrevValues(settings)
+
+                const primitiveForm = settingsToForm(settings)
+                setValues(primitiveForm)
+                setPrevValues(primitiveForm)
             })()
         }
     }, [isTauri, settings])
 
-    const onChange = useCallback((_changes: Partial<ISettings>, values_: ISettings) => {
-        setValues(values_)
-    }, [])
+    const validateSettings = useCallback(
```

Comment:
We should use this to do the validation: https://github.com/openai-translator/openai-translator/blob/41ef88b98227e65a178b43acda2a84e3bbb149b1/src/common/components/Form/form.ts#L49

Reply:
If I put the validation in each FormItem, how should I write the validator to achieve the current rules (mainly that different FormItems will affect each other)? Do you have any suggestions or examples?